### DISCUSSION TITLE
Are Color Ramps and Shadows working properly together?

### DISCUSSION THREAD

**Discussion by aVersionOfReality** [bnpr/Malt#579] [Development]
Here is an NPR Diffuse node using a color ramp with shadows enabled. It is being shadowed by the unselected sphere. The shadowed areas are pure black, but you can see that the Shading is following the color ramp. Is this intended? I would have thought the Shadowed area would also be the 0 end of the color ramp color?

![image](https://github.com/user-attachments/assets/ee3977d5-8b10-46f6-b3f1-38b8a1033b54)

**Comment by pragma37** [comment]
That's what the base color is for, just set the base color to red.

The thing is the ramp is additively evaluated for each light in the scene, if each light contributed light to the surface, even when in shadow or out of radius, the shadow color would get brighter with each light added to the scene. 🙀
